pyorbital/orbital.py

```diff
@@ -159,16 +162,14 @@ def __init__(self, satellite, tle_file=None, line1=None, line2=None):
         self._sgdp4 = _SGDP4(self.orbit_elements)
 
     def __str__(self):
+        """Print the Orbital object state."""
         return self.satellite_name + " " + str(self.tle)
 
     def get_last_an_time(self, utc_time):
-        """Calculate time of last ascending node relative to the
-        specified time
-        """
-
+        """Calculate time of last ascending node relative to the specified time."""
         # Propagate backwards to ascending node
         dt = np.timedelta64(10, 'm')
-        t_old = utc_time
+        t_old = np.datetime64(utc_time)
```

Comment:
FYI in modern python times a datetime object is "supposed to" have a timezone defined on it. If you use this method of conversion here and a python datetime object has a timezone then you'll get a warning:

```
In [13]: now = dt.datetime.now(dt.timezone.utc)

In [14]: np.datetime64(now)
<ipython-input-14-019101e2dae5>:1: UserWarning: no explicit representation of timezones available for np.datetime64
  np.datetime64(now)
Out[14]: np.datetime64('2024-07-17T15:36:14.860486')

```

I *think* you could check if the datetime has a `.tzinfo` and if not then convert it to UTC and then drop the timezone (not sure how to do that yet) before passing it to `np.datetime64` to avoid the warning. Or I guess you could drop any timezone information no matter what because you have defined this interface as expecting time in UTC. :thinking:

Reply:
Good point, thanks! I believe I am handling this now!?